File: old/basic/src/main/java/com/dinesh/android/java/android_lifecycle/view_model/MainActivity.java
package com.dinesh.android.java.android_lifecycle.view_model;

import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import com.dinesh.android.R;

public class MainActivity extends AppCompatActivity {
    private final String TAG = "log_" + getClass().getName().split(getClass().getName().split("\\.")[2] + ".")[1];

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_live_data);
        TextView mTextView = findViewById(R.id.tvNumber);
        Log.e("TAG_", "onCreate: ");

        //  ViewModel
//        MainViewModel myData = new MainViewModel();
        MainViewModel myData = new ViewModelProvider(this).get(MainViewModel.class);
        String myRandomNumber = myData.getNumber();
        mTextView.setText(myRandomNumber);

        Log.i(TAG, "Random Number Set");

    }
}

File: old/basic/src/main/java/com/dinesh/android/kotlin/database/room/v0/MainActivity.kt
package com.dinesh.android.kotlin.database.room.v0

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider


class MainActivity : AppCompatActivity() {
    private val TAG = "log_" + MainActivity::class.java.name.split(MainActivity::class.java.name.split(".").toTypedArray()[2] + ".").toTypedArray()[1]

    private var userViewModel: UserViewModel? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

//        userViewModel = ViewModelProvider(this)[UserViewModel::class.java]
        userViewModel = ViewModelProvider(this).get(UserViewModel::class.java)

        userViewModel!!.allUsers?.observe(this) {
            Log.d(TAG, "onCreate: $it")
        }

        userViewModel!!.insert(User("userName 1", "password123"))
    }

    override fun onDestroy() {
        super.onDestroy()
        userViewModel!!.shutdownExecutor()
    }
}